### PR TITLE
Fix debug

### PR DESCRIPTION
**Pull request by judith-ipac**

### CODE REVIEW

File: src/Query/ADQL_parser.hxx [hunk 1]
```diff
@@ -1,7 +1,14 @@
 #pragma once
 #include <sstream>
 
+#if 0
+#define BOOST_SPIRIT_DEBUG
+#define BOOST_SPIRIT_DEBUG_OUT std::cout
```
Comment (azhangipac) [inline src/Query/ADQL_parser.hxx:6]: Are they referred anywhere?

File: src/Query/ADQL_parser/ADQL_parser/init_column_reference.cxx [hunk 2]
```diff
@@ -81,6 +81,17 @@ void ADQL_parser::init_column_reference() {
     qualifier_column %= (tap_upload | identifier) >> '.' >> identifier;
     qualifier_column.name("qualifier column");
     column_reference %= catalog_schema_qualifier_column | schema_qualifier_column |
-                        qualifier_column | identifier;
+                        qualifier_column | column_name;
```
Comment (azhangipac) [inline src/Query/ADQL_parser/ADQL_parser/init_column_reference.cxx:84]: Wondering if 'column_name' being assigned with 'identifier' already?

File: src/Query/ADQL_parser/ADQL_parser/init_geometry.cxx [hunk 3]
```diff
@@ -75,7 +75,36 @@ void ADQL_parser::init_geometry() {
             ascii::no_case["INTERSECTS"] >> '(' > value_expression > ',' > shape > ')';
     intersects.name("intersects");
 
-    geometry %= ((contains | intersects) >> -(lit('=') >> '1')) |
-                (lit('1') >> '=' >> (contains | intersects));
+    char_flag %= char_('0') | char_('1');
+
+    geometry_left %= char_flag >> lit('=') >> (contains | intersects);
+    geometry_right %= (contains | intersects) >> ((lit('=') > char_flag) | attr('1'));
```
Comment (azhangipac) [inline src/Query/ADQL_parser/ADQL_parser/init_geometry.cxx:81]: ">>"?

**Review by azhangipac — APPROVED** [review]
Looks good to me.

File: src/Query/ADQL_parser.hxx (hunk 1 above)
Comment (judith-ipac) [reply src/Query/ADQL_parser.hxx:6]: RIght now they're enclosed in an "if 0" block, but if the '0' is changed to a '1', then these defines turn on boost::spirit's debugging mechanism.

File: src/Query/ADQL_parser/ADQL_parser/init_column_reference.cxx (hunk 2 above)
Comment (judith-ipac) [reply src/Query/ADQL_parser/ADQL_parser/init_column_reference.cxx:84]: init_columns.cxx has this: `column_name %= identifier;`.

Comment (azhangipac) [reply src/Query/ADQL_parser/ADQL_parser/init_column_reference.cxx:84]: init_columns() is called after init_column_reference(). Order does not matter?

Comment (judith-ipac) [reply src/Query/ADQL_parser/ADQL_parser/init_column_reference.cxx]: RIght, the order in which the init_XXX() functions are called in ADQL_parser.cxx does not matter.  I scrambled them completely and it had no effect on either the libadql test suite or the query_server-level regression tests.  But I've reverted the change for now and will perhaps do a larger-scale cleanup/reorganization of the initialization code later on.

**Pull request merged**